airflow/www/static/js/connection_form.js

```diff
@@ -140,6 +140,55 @@ $(document).ready(function () {
     }
   }
 
+  $.fn.serializeObject = function () {
+    const o = {};
+    const a = this.serializeArray();
+    $.each(a, function () {
+      if (this.name === 'csrf_token') {
+        // dont send csrf token
+      } else if (this.name === 'conn_id') {
+        o.connection_id = this.value;
+      } else if (this.value !== '') {
+        o[this.name] = this.value;
+      }
+    });
+    return o;
+  };
```

Comment:
I think this functionality is built in to jQuery.

/cc @bbovenzi

Reply:
Yes, it should be just `.serialize()` https://api.jquery.com/serialize/

Reply:
@bbovenzi yeah, I tried `serialize` but it was giving me a string like `key1=val1&key2=val2` which again I need to convert into a proper json object.

Reply:
Ah gotcha.

Another option would be to convert the API endpoint to _also_ accept www-form-encoded data?

Reply:
I would like to keep the APIs simple and not driven by the needs of UI.